packages/binding-http/src/http-server.ts

```diff
@@ -205,113 +209,126 @@ export default class HttpServer implements ProtocolServer {
       console.debug("[binding-http]",`HttpServer on port ${this.getPort()} exposes '${thing.title}' as unique '/${urlPath}'`);
       this.things.set(urlPath, thing);
 
-      // fill in binding data
-      for (let address of Helpers.getAddresses()) {
-        for (let type of ContentSerdes.get().getOfferedMediaTypes()) {
+      if (this.baseUri !== undefined) {
+        let base: string = this.baseUri.concat("/", encodeURIComponent(urlPath))
+        console.info("[binding-http]", "HttpServer TD hrefs using baseUri " + this.baseUri)
+        this.addEndpoint(thing, tdTemplate, base )
+      } else {
+        // fill in binding data
+        for (let address of Helpers.getAddresses()) {
           let base: string = this.scheme + "://" + address + ":" + this.getPort() + "/" + encodeURIComponent(urlPath);
 
-          if (true) { // make reporting of all properties optional?
-            // check for readOnly/writeOnly for op field and whether there are any properties at all
-            let allReadOnly = true;
-            let allWriteOnly = true;
-            let anyProperties = false;
-            for (let propertyName in thing.properties) {
-              anyProperties = true;
-              if (!thing.properties[propertyName].readOnly) {
-                allReadOnly = false;
-              } else if (!thing.properties[propertyName].writeOnly) {
-                allWriteOnly = false;
-              }
-            }
-            if (anyProperties) {
-              let href = base + "/" + this.ALL_DIR + "/" + encodeURIComponent(this.ALL_PROPERTIES);
-              let form = new TD.Form(href, type);
-              if (allReadOnly) {
-                form.op = ["readallproperties", "readmultipleproperties"];
-              } else if (allWriteOnly) {
-                form.op = ["writeallproperties", "writemultipleproperties"];
-              } else {
-                form.op = ["readallproperties", "readmultipleproperties", "writeallproperties", "writemultipleproperties"];
-              }
-              if (!thing.forms) {
-                thing.forms = [];
-              }
-              thing.forms.push(form);
-            }
-          }
+          this.addEndpoint(thing, tdTemplate, base)
+          // media types
+        } // addresses
 
+        if (this.scheme === "https") {
+          this.fillSecurityScheme(thing)
+        }
+
+      } // running
+
+      return new Promise<void>((resolve, reject) => {
+        resolve();
+      });
+    }
+  }
+
+  public addEndpoint(thing: ExposedThing, tdTemplate: WoT.ThingDescription, base: string) {
+      for (let type of ContentSerdes.get().getOfferedMediaTypes()) {
+
+        if (true) { // make reporting of all properties optional?
```

Comment:
Do we still need this `if`? 🤔 since we are refactoring I'd remove it.

Reply:
removed